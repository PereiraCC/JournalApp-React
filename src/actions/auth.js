import { googleAuthProvider, getAuth, signInWithPopup, createUserWithEmailAndPassword, updateProfile, signInWithEmailAndPassword } from '../firebase/firebase-config';
import { types } from '../types/types';
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = (email, password) => {
    return ( dispatch ) => {

        dispatch( startLoading() );

        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password)
            .then(({ user }) => {
                dispatch( login( user.uid, user.displayName ) );
                
                dispatch( finishLoading() );
            })
            .catch( e => {
                dispatch( finishLoading() );
                console.log(e);
            });      
    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {

    return (dispatch ) => {
        const auth = getAuth();
        createUserWithEmailAndPassword(auth, email, password )
            .then( async ({ user }) => {

                await updateProfile( auth.currentUser, { 
                    displayName: name
                });

                dispatch(
                    login( user.uid, user.displayName )
                );
            })
            .catch( e => {
                console.log(e);
            })
    }

}

export const startGoogleLogin = () => {
    return ( dispatch ) => {

        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then( ({ user }) => {
                dispatch(
                    login( user.uid, user.displayName )
                )
            });
    }
}

export const login = ( uid, displayName ) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }
});

// export const logout = () => ({
//     type: types.logout,
// });



