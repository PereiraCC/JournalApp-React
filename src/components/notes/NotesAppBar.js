import { useDispatch, useSelector } from 'react-redux';

import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

    const { active } = useSelector( state => state.notes );
    const dispatch = useDispatch();

    const handleSaveNote = () => {
        dispatch( startSaveNote( active ) );
    }

    const handleUploadPicture = () => {
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) => {
        const file = e.target.files[0];
        if( file ) {
            dispatch( startUploading( file ) );
        }
    }

    return (
        <div className="notes__appbar">
            <span>29 January 2022</span>

            <input 
                id="fileSelector"
                type="file"
                name="file"
                style={ { display: 'none' } }
                onChange={ handleFileChange }
            />

            <div>
                <button 
                    className="btn"
                    onClick={ handleUploadPicture }    
                >
                    Picture
                </button>
                
                <button 
                    className="btn"
                    onClick={ handleSaveNote }
                >
                    Save
                </button>
            </div>
        </div>
    );
};
