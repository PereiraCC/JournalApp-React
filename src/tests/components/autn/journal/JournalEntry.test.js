import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { activeNote } from '../../../../actions/notes';
import { JournalEntry } from '../../../../components/journal/JournalEntry';

const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

const initialState = {};

let store = mockStore( initialState );
store.dispatch = jest.fn();

const note = {
    id: 1,
    title: 'Note title',
    body: 'Note body',
    date: '2020-01-01',
    url: 'https://image.com/image.jpg',
}

const wrapper = mount( 
    <Provider store={ store} >
        <JournalEntry { ...note }/>
    </Provider>
);


describe('<JournalEntry /> ', () => { 

    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should active note', () => { 

        wrapper.find('.journal__entry').prop('onClick')();
        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( note.id, { ...note } )
        );

    });

});