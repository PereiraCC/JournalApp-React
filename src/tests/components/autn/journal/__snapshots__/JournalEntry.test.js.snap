// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalEntry />  should show correctly 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://image.com/image.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Note title
    </p>
    <p
      className="journal__entry-content"
    >
      Note body
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      1st
    </h4>
  </div>
</div>
`;
